components/KiboPayment.vue

```diff
@@ -0,0 +1,200 @@
+                </SfIcon>
+              </span>
+            </SfButton>
+            <label for="cardNumber" class="sf-input__label">{{ $t("Card number") }}</label>
+          </div>
+          <div class="sf-input__error-message"><div class="display-none"></div></div>
+        </div>
+        <SfInput
+          v-model="creditCardFormData.expiryDate"
+          :label="$t('Exp. Date MM/YY')"
+          name="expiryDate"
+          class="credit-card-form__element"
+          :required="true"
+          @input="updateField()"
+        />
+        <div class="credit-card-form__element custom-form-element sf-input">
+          <div class="sf-input__wrapper">
+            <input
+              v-model="creditCardFormData.securityCode"
+              type="password"
+              name="securityCode"
+              class="custom-form-element__input"
+              :required="true"
+              @input="updateField()"
+            />
+            <SfButton class="custom-form-element__button sf-button--pure">
+              <span class="custom-form-element__icon">
+                <SfIcon size="1.25rem">
+                  <font-awesome-icon icon="question-circle" class="fa-icon" />
+                </SfIcon>
+              </span>
+            </SfButton>
+            <label for="securityCode" class="sf-input__label">{{ $t("Security Code") }}</label>
+          </div>
+          <div class="sf-input__error-message"><div class="display-none"></div></div>
+        </div>
+        <SfCheckbox
+          v-model="creditCardFormData.cardKeep"
+          name="keepcard"
+          :disabled="!paymentMethod"
+          label="Save payment method"
+          class="credit-card-form__checkbox"
+          @change="updateField()"
+        />
+      </div>
+    </template>
+  </SfPayment>
+</template>
+<script lang="ts">
+import { defineComponent } from "@vue/composition-api"
+import { SfPayment, SfRadio, SfInput, SfCheckbox, SfIcon, SfButton } from "@storefront-ui/vue"
+import { ref } from "@nuxt/bridge/dist/runtime"
+
+export default defineComponent({
+  name: "KiboPayment",
+  components: {
+    SfPayment,
+    SfRadio,
+    SfInput,
+    SfCheckbox,
+    SfIcon,
+    SfButton,
+  },
+  props: {
+    paymentMethods: {
+      type: Array,
+      default: () => [
+        {
+          label: "Credit / Debit Card",
+          value: "creditCard",
+        },
+      ],
+    },
+    years: {
+      type: Array,
+      default: () => [],
+    },
+    months: {
+      type: Array,
+      default: () => [],
+    },
+  },
+
+  setup(_props, _context) {
+    const paymentMethod = ref(null)
+    const creditCardFormData = ref({
+      cardNumber: "",
+      expiryDate: "",
+      securityCode: "",
+      cardKeep: false,
+      cardType: "",
+    })
+
+    const updateField = () => {
+      _context.emit("input", {
+        ...creditCardFormData,
+      })
+      creditCardFormData.value.cardType = paymentMethod
```

Comment:
Add the library, credit-card-type, to determine a cards type based on number

Reply:
Added the library